### PR TITLE
updated release process

### PR DESCRIPTION
**Pull request by keith-turner**

### CODE REVIEW

File: pages/release-process.md
Comment (ctubbsii) [inline pages/release-process.md]: Could mention that instructions are provided at top of KEYS file on how to do this.

File: pages/release-process.md
Comment (ctubbsii) [inline pages/release-process.md]: Should also link to Maven page on securing your credentials: https://maven.apache.org/guides/mini/guide-encryption.html

File: pages/release-process.md
Comment (ctubbsii) [inline pages/release-process.md]: Make bullet points consistent with terminating punctuation.
* Some lines have a period.
* Some don't

File: pages/release-process.md
Comment (ctubbsii) [inline pages/release-process.md]: I don't think this is the correct command to start the gpg-agent. The previously documented command actually starts a gpg-agent. This may not be necessary if you're running inside a desktop environment, like GNOME or KDE, which provides a gpg-agent for you (gnome-keyring-daemon provides this in GNOME, for example).

What this gpg-connect-agent command does is clears the currently running gpg-agent's cache. The create-release-candidate script does this to help guarantee that the timer for your cached passphrase is reset before a potentially long-running build. The user still must ensure that their gpg-agent is configured with a sufficiently long timeout so that the cached passphrase won't expire during the build.

File: pages/release-process.md
Comment (ctubbsii) [inline pages/release-process.md]: Might be worth mentioning that `0` is a good choice for the `RCV` for testing the release candidate process itself or for branching a pre-release candidate for testing. It's not a strictly necessary convention, but I've found it to be a useful one to make sure I don't get confused between an actual release candidate and a test one, not intended for voting.

File: pages/release-process.md
Comment (ctubbsii) [inline pages/release-process.md]: This really shouldn't be necessary. If we ever have to do this, it means we haven't done a good enough job with the build to auto-substitute these version numbers based on the current build. If there exist any such cases currently, we should file issues to automate them. This step should never be necessary during release candidate creation.

File: pages/release-process.md
Comment (ctubbsii) [inline pages/release-process.md]: This step also creates two commits in the current branch, and creates a tag of the first commit of these two. That tag is used by `release:prepare`, but should be deleted afterwards.

File: pages/release-process.md
Comment (ctubbsii) [inline pages/release-process.md]: The staging repository is viewable by all committers on the project, not just its creator. Also, it's useful to put in a description of the repo when one closes it. Something like "Apache Fluo (incubating) 1.1.0-rc3".

File: pages/release-process.md
Comment (ctubbsii) [inline pages/release-process.md]: It's important that the staging repo is *never* promoted/released until after a vote is successful, because that will push its contents to Maven central as though it were an official release. This can't be undone, and the only correct course of action (if the vote fails) is to skip over that release number for the next release candidate.

File: pages/release-process.md
Comment (ctubbsii) [inline pages/release-process.md]: This should already be the current checkout at this point.

File: pages/release-process.md
Comment (ctubbsii) [inline pages/release-process.md]: ... and one commit ahead of the branch point.

File: pages/release-process.md
Comment (ctubbsii) [inline pages/release-process.md]: The "-next" branch should be merged into (or become, if one didn't already exist) the maintenance branch for the version which was just released (if a maintenance branch is desired), and then that maintenance branch should be merged forward into maintenance branches for newer releases, until eventually merging into master. (not sure how to make that more succinct)

File: pages/release-process.md
Comment (ctubbsii) [inline pages/release-process.md]: Again, useful to have a comment: "Vote failed for Apache Fluo 1.1.0-rc3" or similar for dropping old staging repos; "Apache Fluo 1.1.0" for released repos.

File: pages/release-process.md
Comment (ctubbsii) [inline pages/release-process.md]: Users may wish to do `git config --global --get gpg.program || git config --global --add gpg.program gpg2`
This is because some systems which have both `gpg` and `gpg2` installed may not function correctly. (See https://bugzilla.redhat.com/show_bug.cgi?id=568406)

File: pages/release-process.md
Comment (ctubbsii) [inline pages/release-process.md:127]: Are you sure this is correct? I thought you needed the `--tags` option to push tags unless you specify the fully-qualified ref name: `git push origin refs/tags/release-1.0:refs/tags/release-1.0`

I think what your command will do is try to push a new branch from the currently checked-out branch/commit to the remote with the name `rel/fluo-<releaseVersion>`.

Might be a good idea to document testing it with `--dry-run` first.

File: pages/release-process.md
Comment (ctubbsii) [inline pages/release-process.md]: I didn't know spaces worked in markdown link references!

**Review by ctubbsii — CHANGES_REQUESTED** [review]
This is good stuff. This page was *way* out of date. I made a few suggestions.

File: pages/release-process.md
Comment (keith-turner) [reply pages/release-process.md]: I was just making it crystal clear.

File: pages/release-process.md
Comment (keith-turner) [reply pages/release-process.md:127]: I think I tried this pushing to my fork, but not completely sure.  I'll test again.

File: pages/release-process.md
Comment (ctubbsii) [reply pages/release-process.md]: In the latest version, this wording is still incorrect.

File: pages/release-process.md
Comment (ctubbsii) [inline pages/release-process.md]: ahead of the branch point :)

File: pages/release-process.md
Comment (ctubbsii) [inline pages/release-process.md]: More accurate:
s/only to you/to you only/

It's visible to all committers on the project after logging in, but it's not necessary to specify all that. Shifting the word "only" to the right twice makes the statement correct.

**Comment by ctubbsii** [comment]
Recent changes look good. I see you incorporated pretty much all of my feedback from my first review. Thanks for that! I reiterated two more minor issues from my first review that I think should be addressed, but otherwise, looks good to me.

File: pages/release-process.md
Comment (keith-turner) [reply pages/release-process.md]: I decided to drop the whole bit.  Even it we make it correct, I feel its not very helpful/useful info.

**Comment by keith-turner** [comment]
Still need to document putting the source and bin tarballs in svn so they end up on the mirrors.  Also need to document what needs to be done to remove older releases from mirrors.

File: pages/release-process.md
Comment (keith-turner) [reply pages/release-process.md:127]: I used this command to create https://github.com/apache/incubator-fluo/releases/tag/rel%2Ffluo-1.1.0-incubating

**Review by ctubbsii — APPROVED** [review]
Everything looks good to me. Anything new should probably start in a new PR. This one is getting hard to follow with all the incremental updates, especially when comparing to the original, due to the dos2unix line ending change.

**Pull request merged**
